src/richie/plugins/lti_consumer/factories.py

```diff
@@ -14,7 +17,16 @@ class LTIConsumerFactory(factory.django.DjangoModelFactory):
     class Meta:
         model = LTIConsumer
 
-    url = factory.Faker("url")
-    lti_provider_id = "lti_provider_test"
-    oauth_consumer_key = None
-    shared_secret = None
+    lti_provider_id = factory.fuzzy.FuzzyChoice(
+        getattr(settings, "RICHIE_LTI_PROVIDERS", {}).keys()
+    )
+
+    @factory.lazy_attribute
+    def url(self):
+        """Generate a random url in accordance with the LTI provider."""
```

Comment:
```suggestion
        """Generates a random url in accordance with the LTI provider."""
```
A master told me once, Return**s** for the same context so I think it should be Generates